Sim_To_DuT_Interface/DuT_Connectors/DuTConnector.h

```diff
@@ -40,23 +43,43 @@ namespace sim_interface::dut_connector {
         explicit DuTConnector(std::shared_ptr<SharedQueue<SimEvent>> queueDuTToSim,
                               const sim_interface::dut_connector::ConnectorConfig &config);
 
+        ~DuTConnector();
+
         // return some basic information from the connector
         virtual ConnectorInfo getConnectorInfo();
 
         // handle event from the simulation async
-        virtual void handleEvent(const SimEvent &simEvent);
+        void handleEvent(const SimEvent &simEvent);
 
-// send the event to the simulation
+        // send the event to the simulation
         void sendEventToSim(const SimEvent &simEvent);
 
     protected:
+        virtual void handleEventSingle(const SimEvent &simEvent) {};
```

Comment:
Ich würde gerne "virtuall handleEvent" behalten und eine andere Methode erstellen die dann den Aktuellen Code von handleEvent ausführt erstellen.
Falls nicht, muss noch der Code in CANConnector angepasst werden!

Reply:
Aber vom Interface aus muss ja der aktuelle Code ausgeführt werden unabhängig vom Connector. Also würde ich sagen müssten wir im CANConnector den Code in das Single verschieben

Reply:
Ok